### PR TITLE
Make compatible with current nightly

### PR DESCRIPTION
**Pull request by ctz**
*Description of changes:*
I learned about this tool today at eurorust -- this PR aims to make it compatible with current nightly.

Prior to this PR, the output is:
```
$ cargo install --locked cargo-check-external-types
(...)
$ cargo +nightly check-external-types --config external-types.toml
Running rustdoc to produce json doc output...
[/home/jpixton/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-check-external-types-0.1.7/src/main.rs:101] err = Error {
    context: "error at /home/jpixton/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-check-external-types-0.1.7/src/main.rs:171:14",
    source: "The version of rustdoc being used produces JSON format version 27, but this tool requires format version 26. This can happen if the locally installed version of rustdoc doesn't match the rustdoc JSON types from the `rustdoc-types` crate.\n\nIf this occurs with the latest Rust nightly and the latest version of this tool, then this is a bug, and the tool needs to be upgraded to the latest format version.\n\nOtherwise, you'll need to determine a Rust nightly version that matches this tool's supported format version (or vice versa).",
}
error at /home/jpixton/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-check-external-types-0.1.7/src/main.rs:171:14: The version of rustdoc being used produces JSON format version 27, but this tool requires format version 26. This can happen if the locally installed version of rustdoc doesn't match the rustdoc JSON types from the `rustdoc-types` crate.

If this occurs with the latest Rust nightly and the latest version of this tool, then this is a bug, and the tool needs to be upgraded to the latest format version.

Otherwise, you'll need to determine a Rust nightly version that matches this tool's supported format version (or vice versa).
```

When taking rustdoc-types 0.23 I took the liberty of tracking the naming change around type aliases. That changes the public interface of this tool in a minor way.

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jdisanti — COMMENTED** [review]
Hey, thanks a ton for tackling this! ❤️ 

It looks like just a clippy lint needs to be fixed to pass CI. Once that passes, I'll get this merged and release. The changes look good.

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "nightly-2023-05-31"
+channel = "nightly-2023-10-10"
```
[the remark this answers is not in the document]
Comment (ctz) [reply rust-toolchain.toml:2]: updated

**Comment by ctz** [comment]
> It looks like just a clippy lint needs to be fixed to pass CI.

Added a commit which addresses the new lint

**Review by jdisanti — APPROVED** [review]

**Comment by jdisanti** [comment]
Thank you!

**Comment by jdisanti** [comment]
It's published to crates.io now.

**Pull request merged**